src/coreComponents/constitutive/permeability/ParallelPlatesPermeability.hpp

```diff
@@ -0,0 +1,140 @@
+
+#ifndef GEOSX_CONSTITUTIVE_PERMEABILITY_PARALLELPLATESPERMEABILITY_HPP_
+#define GEOSX_CONSTITUTIVE_PERMEABILITY_PARALLELPLATESPERMEABILITY_HPP_
+
+#include "constitutive/permeability/PermeabilityBase.hpp"
+
+
+namespace geosx
+{
+namespace constitutive
+{
+
+class ParallelPlatesPermeabilityUpdate : public PermeabilityBaseUpdate
+{
+public:
+
+  ParallelPlatesPermeabilityUpdate( arrayView3d< real64 > const & permeability,
+                                    arrayView3d< real64 > const & dPerm_dAperture )
+    : PermeabilityBaseUpdate( permeability ),
+    m_dPerm_dAperture( dPerm_dAperture )
+  {}
+
+  /// Default copy constructor
+  ParallelPlatesPermeabilityUpdate( ParallelPlatesPermeabilityUpdate const & ) = default;
+
+  /// Default move constructor
+  ParallelPlatesPermeabilityUpdate( ParallelPlatesPermeabilityUpdate && ) = default;
+
+  /// Deleted copy assignment operator
+  ParallelPlatesPermeabilityUpdate & operator=( ParallelPlatesPermeabilityUpdate const & ) = delete;
+
+  /// Deleted move assignment operator
+  ParallelPlatesPermeabilityUpdate & operator=( ParallelPlatesPermeabilityUpdate && ) = delete;
+
+  GEOSX_HOST_DEVICE
+  GEOSX_FORCE_INLINE
+  void compute( real64 const & effectiveAperture,
+                arraySlice1d< real64 > const & permeability,
+                arraySlice1d< real64 > const & dPerm_dAperture ) const;
+
+  GEOSX_HOST_DEVICE
+  GEOSX_FORCE_INLINE
+  void update( localIndex const k,
+               localIndex const q,
+               real64 const & effectiveAperture ) const
+  {
+    compute( effectiveAperture,
+             m_permeability[k][q],
+             m_dPerm_dAperture[k][q] );
+  }
+private:
+  arrayView3d< real64 > m_dPerm_dAperture;
+
+};
+
+
+class ParallelPlatesPermeability : public PermeabilityBase
+{
+public:
+  ParallelPlatesPermeability( string const & name, Group * const parent );
+
+  virtual ~ParallelPlatesPermeability() override;
+
+  std::unique_ptr< ConstitutiveBase > deliverClone( string const & name,
+                                                    Group * const parent ) const override;
+
+  virtual void allocateConstitutiveData( dataRepository::Group & parent,
+                                         localIndex const numConstitutivePointsPerParentIndex ) override;
+
+  static string catalogName() { return "ParallelPlatesPermeability"; }
+
+  virtual string getCatalogName() const override { return catalogName(); }
+
+  /// Type of kernel wrapper for in-kernel update
+  using KernelWrapper = ParallelPlatesPermeabilityUpdate;
+
+  /**
+   * @brief Create an update kernel wrapper.
+   * @return the wrapper
+   */
+  KernelWrapper createKernelWrapper()
+  {
+    return KernelWrapper( m_permeability,
+                          m_dPerm_dAperture );
+  }
+
+
+  struct viewKeyStruct : public PermeabilityBase::viewKeyStruct
+  {
+    static constexpr char const * dPerm_dApertureString() { return "dPerm_dAperture"; }
+  } viewKeys;
+
+private:
+  array3d< real64 > m_dPerm_dAperture;
```

Comment:
Yes, any suggestion about how to do this would be very useful. It would be great if the solver could throw an error if someone tried to use a constitutive model that does not make sense. 

The reason why I am not registering all variables in the base class and I am not enforcing a generic `update ` function is to avoid carrying around useless data. For example, I think that if we run a flow-only problem there should not be the need for storing derivatives of the permeability w.r.t. to mechanics unknowns. As a matter of fact I was trying to make the flow solver completely agnostic of all mechanics related quantities.